src/Apollo/NcrReactionsProjector.php

```diff
@@ -0,0 +1,216 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Apollo;
+
+use Aws\DynamoDb\DynamoDbClient;
+use Gdbots\Ncr\Aggregate;
+use Gdbots\Ncr\Event\NodeProjectedEvent;
+use Gdbots\Ncr\Exception\NodeNotFound;
+use Gdbots\Ncr\Ncr;
+use Gdbots\Ncr\NcrSearch;
+use Gdbots\Ncr\Repository\DynamoDb\NodeTable;
+use Gdbots\Ncr\Repository\DynamoDb\TableManager;
+use Gdbots\Pbj\Message;
+use Gdbots\Pbj\MessageResolver;
+use Gdbots\Pbj\WellKnown\Microtime;
+use Gdbots\Pbj\WellKnown\NodeRef;
+use Gdbots\Pbjx\DependencyInjection\PbjxProjector;
+use Gdbots\Pbjx\EventSubscriber;
+use Gdbots\Pbjx\Pbjx;
+use Gdbots\Schemas\Ncr\Enum\NodeStatus;
+
+class NcrReactionsProjector implements EventSubscriber, PbjxProjector
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'gdbots:ncr:mixin:node.projected'      => 'onNodeProjected',
+            'triniti:apollo:event:reactions-added' => 'onReactionsAdded',
+        ];
+    }
+
+    public function __construct(
+        protected DynamoDbClient $client,
+        protected TableManager $tableManager,
+        protected Ncr $ncr,
+        protected NcrSearch $ncrSearch,
+        protected bool $enabled = true
+    ) {
+    }
+
+    public function onNodeProjected(NodeProjectedEvent $pbjxEvent): void
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $node = $pbjxEvent->getNode();
+        $nodeRef = $node->generateNodeRef();
+        $lastEvent = $pbjxEvent->getLastEvent();
+
+        if (!$this->hasReactions($nodeRef)) {
+            return;
+        }
+
+        if (NodeStatus::DELETED->value === $node->fget('status')) {
+            $context = ['causator' => $lastEvent];
+            $reactionsRef = $this->createReactionsRef($nodeRef);
+            $this->ncr->deleteNode($reactionsRef, $context);
+            $this->ncrSearch->deleteNodes([$reactionsRef], $context);
+            return;
+        }
+
+        $reactions = $this->getReactions($nodeRef, $lastEvent) ?? $this->createReactions($nodeRef);
+
+        $this->mergeNode($node, $reactions);
+        $this->projectNode($reactions, $lastEvent, $pbjxEvent::getPbjx());
```

Comment:
@bhsiao55 this doesnt cause an infinite loop because reactions do not themselves have reactions? is that right?

Reply:
@bhsiao55 doesn't need to do anything with ncrSearch (add or delete), this class shouldn't have that service.

Reply:
> @bhsiao55 this doesnt cause an infinite loop because reactions do not themselves have reactions? is that right?

@efjacobson reactions node do not have reactions, but I'm unclear where the infinite loop could be